player/lua/console.lua

```diff
@@ -811,30 +813,68 @@ function build_completers()
     return completers
 end
 
--- Use 'list' to find possible tab-completions for 'part.'
--- Returns a list of all potential completions and the longest
--- common prefix of all the matching list items.
-function complete_match(part, list)
-    local completions = {}
+-- Find the longest common case-sensitive prefix of the entries in "list".
+local function find_common_prefix(part, list)
     local prefix = nil
 
     for _, candidate in ipairs(list) do
-        if candidate:sub(1, part:len()) == part then
-            if prefix and prefix ~= candidate then
-                local prefix_len = part:len()
-                while prefix:sub(1, prefix_len + 1)
-                       == candidate:sub(1, prefix_len + 1) do
-                    prefix_len = prefix_len + 1
-                end
-                prefix = candidate:sub(1, prefix_len)
-            else
-                prefix = candidate
+        if prefix and prefix ~= candidate then
+            local prefix_len = part:len()
+            while prefix:sub(1, prefix_len + 1) == candidate:sub(1, prefix_len + 1) do
+                prefix_len = prefix_len + 1
             end
+            prefix = candidate:sub(1, prefix_len)
+        else
+            prefix = candidate
+        end
+    end
+
+    return prefix
+end
+
+-- Return the entries of "list" beginning with "part" and the longest common
+-- prefix of the matches.
+local function complete_match(part, list)
+    local completions = {}
+
+    for _, candidate in pairs(list) do
+        if candidate:sub(1, part:len()) == part then
             completions[#completions + 1] = candidate
         end
     end
 
-    return completions, prefix
+    local prefix = find_common_prefix(part, completions)
+
+    if opts.case_sensitive then
+        return completions, prefix
+    end
+
+    completions = {}
+    local lower_case_completions = {}
+
+    for _, candidate in pairs(list) do
+        if candidate:sub(1, part:len()):lower() == part:lower() then
```

Comment:
Doing `part:lower()` on each iteration ain't great, maybe store it in  a variable instead?

I'm sure it doesn't really matter performance wise in this case, but I'm still bothered by it :P